### PR TITLE
Events are also now tracked to an external API if configured

### PR DESCRIPTION
**Pull request by aicamargo**
When triggering an event like a deploy or rollback, an HTTP request is sent to the configured REST endpoint for processing. This allows for more detailed tracking of events than statsd currently provides. 

@dnfehren @jpunzel @danrjohnson @blakesmith

### CODE REVIEW

**Comment by dnfehren** [comment]
Works for me. Making the graphite reporting function and the kudzu reporting function an instance of some kind of `external_report` class might be cool if we were going ship this stuff a lot of different places, but might be too much work for what it's worth.

**Comment by danrjohnson** [comment]
This looks good to me

File: mendel/__init__.py [hunk 1]
```diff
@@ -11,7 +11,6 @@
 from mendel.util import create_host_task
 from mendel.util import ConfigMissingError
 from mendel.util import load_mendel_config
-from mendel.version import __version__
```
Comment (stantonk) [inline mendel/__init__.py]: ???

File: mendel/conf.py [hunk 2]
```diff
@@ -37,3 +37,4 @@ def __init__(self):
         self.NEXUS_PORT = int(os.environ.get('MENDEL_NEXUS_PORT', self._parser.getint_or('nexus', 'port', 0)))
         self.NEXUS_REPOSITORY = os.environ.get('MENDEL_NEXUS_REPOSITORY', self._parser.get_or('nexus', 'repository'))
         self.GRAPHITE_HOST = os.environ.get('MENDEL_GRAPHITE_HOST', self._parser.get_or('graphite', 'host'))
+        self.TRACK_EVENT_ENDPOINT = os.environ.get('TRACK_EVENT_ENDPOINT', self._parser.get_or('api', 'track_event'))
```
Comment (stantonk) [inline mendel/conf.py]: prefix env vars w/ `MENDEL`

File: mendel/core.py [hunk 3]
```diff
@@ -685,6 +692,27 @@ def _track_event(self, event):
...
+    def _track_event_api(self, event):
+        """
+        Track who deployed what service and what the release dir is to an external REST API
+        """
+        if not self._track_event_endpoint:
+            print red('Unable to track deployment event in custom api, no api endpoint configured in ~/.mendel.conf')
+            return
+
+        data = {
+            'service': self._api_service_name or self._service_name,
```
Comment (stantonk) [inline mendel/core.py]: why do we have 2 different attributes for the name of a service?

Comment (stantonk) [inline mendel/core.py]: if this is truly needed, shouldn't the report to graphite also be updated? see line 685

File: mendel/core.py [hunk 4]
```diff
@@ -685,6 +692,27 @@ def _track_event(self, event):
...
+            'host': env.host_string,
+            'deployer': getpass.getuser(),
+            'event': event
+        }
+
+        url = 'http://%s' % self._track_event_endpoint
+
+        r = requests.post(url, data)
```
Comment (stantonk) [inline mendel/core.py]: it's possible this throws an exception, probably want to catch and swallow w/ a warning that tracking the event failed. I should have done that above as well. Also, why introducing python `requests` library? just adding additional dependencies to mendel, can use urllib2 as above

File: mendel/version.py [hunk 5]
```diff
@@ -1 +1 @@
-__version__ = '0.6.2'
```
Comment (stantonk) [inline mendel/version.py]: we should probably increment versions after something is merged so we don't risk 2 PR's changing to the same version. also, FYI, mendel is now published to central PyPI as `fabric-mendel`, see: https://pypi.python.org/pypi/fabric-mendel

**Pull request merged**